scripts/dynassault.lua

```diff
@@ -197,14 +184,28 @@ end
 --------------------------------------------------------------------------------
 
 local function RestoreTorsoAim()
-	Signal(SIG_RESTORE_TORSO)
-	SetSignalMask(SIG_RESTORE_TORSO)
-	Sleep(RESTORE_DELAY)
 	Turn(torso, y_axis, restoreHeading, TORSO_SPEED_YAW)
 end
 
+local function RestoreTorsoAfterDelay(sleepTime)
+	local counter = (sleepTime or RESTORE_DELAY)
+	while true do
+		if counter > 0  and not Spring.GetUnitIsStunned(unitID) then
+			counter = counter - 100
+		end
+		if resetRestoreTorso then
+			resetRestoreTorso = false
+			counter = RESTORE_DELAY
```

Comment:
Shouldn't it use the supplied sleep time?
```suggestion
			counter = (sleepTime or RESTORE_DELAY)
```
Although the time is never actually supplied so it's somewhat moot.